src/zsys.c

```diff
@@ -980,14 +980,35 @@ zsys_udp_new (bool routable)
     //  We haven't implemented multicast yet
     assert (!routable);
     SOCKET udpsock;
+    int type = SOCK_DGRAM;
+#if defined ZMQ_HAVE_SOCK_CLOEXEC
```

Comment:
this is not defined anywhere - the logic to detect that needs to be added to zproject

Reply:
I don't think that it's worth to add autoconf machinery for `SOCK_CLOEXEC`, it's just a macro. I fixed the things in PR#1901

Reply:
BTW: thanks for a review!

Reply:
It is necessary - the definition might be available, but it might not work. That's the case on GNU/Hurd for example. See libzmq/acinclude.m4 and libzmq/builds/cmake/Modules/ZMQSourceRunChecks.cmake